src/modules/navigator/mission.cpp

```diff
@@ -1684,6 +1684,13 @@ Mission::need_to_reset_mission(bool active)
 	} else if (_navigator->get_vstatus()->arming_state == vehicle_status_s::ARMING_STATE_ARMED && active) {
 		/* mission is running, need reset after disarm */
 		_need_mission_reset = true;
+
+	} else if (_navigator->get_mission_result()->finished
```

Comment:
We have to be a little careful here because other parts of navigator set the mission result as well.

Reply:
could you help to give some prompt which we need to protect, do we need to confirm vehicle in loiter mode before we reset the mission?

Reply:
@jinchengde we need to make sure that the next state after the mission ends does not crash the vehicle. Example: make sure the vehicle changes to HOLD or to LAND.

Reply:
OK，I will test for that, and I wish to have another PR for more modify

Reply:
@TSC21 I had test in SITL as below
1. automatic takeoff and mission
2. mission end, the vehicle start to circle around the last waypoint
3. change the mode to hold, the vehicle stay current altitude and circle around
4. change the mode to mission, the vehicle fly to the first waypoint
so there is no crash after we change the mode after the mission ends